database/consensus.go

```diff
@@ -44,6 +44,9 @@ func (db *Db) GetLastBlock() (*dbtypes.BlockRow, error) {
 // GetLastBlockHeight returns the last block height stored inside the database
 func (db *Db) GetLastBlockHeight() (int64, error) {
 	block, err := db.GetLastBlock()
+	if block == nil {
+		return 0, fmt.Errorf("block are empty")
+	}
```

Comment:
`nil` checks should always be applied **after** error checks. So this should be 

```golang
if err != nil {
	return 0, err
}
if block == nil {
	return 0, fmt.Errorf("blocks are empty")
}
```
The same applied everywhere else